File: pulse-react-native-otel/src/PulseLogger.ts
import { PulseLogLevel } from './PulseLogLevel';

const TAG = '[PulseSDK]';

let currentLevel: PulseLogLevel = PulseLogLevel.NONE;

export const PulseLogger = {
  setLevel(level: PulseLogLevel): void {
    currentLevel = level;
  },

  getLevel(): PulseLogLevel {
    return currentLevel;
  },

  verbose(message: string): void {
    if (currentLevel <= PulseLogLevel.VERBOSE) {
      console.debug(`${TAG} ${message}`);
    }
  },

  debug(message: string): void {
    if (currentLevel <= PulseLogLevel.DEBUG) {
      console.debug(`${TAG} ${message}`);
    }
  },

  info(message: string): void {
    if (currentLevel <= PulseLogLevel.INFO) {
      console.log(`${TAG} ${message}`);
    }
  },

  warn(message: string): void {
    if (currentLevel <= PulseLogLevel.WARN) {
      console.warn(`${TAG} ${message}`);
    }
  },

  error(message: string): void {
    if (currentLevel <= PulseLogLevel.ERROR) {
      console.error(`${TAG} ${message}`);
    }
  },
};

File: pulse-ios-otel/Sources/PulseLogging/PulseLogger.swift
/*
 * Copyright The Pulse Authors
 * SPDX-License-Identifier: Apache-2.0
 *
 * Central logging for the Pulse SDK. All modules should use PulseLogger
 * so logs have a consistent tag and obey the runtime log level.
 */

import Foundation

package enum PulseLogger {
    private static let tag = "PulseSDK"

    package static var currentLevel: PulseLogLevel = .none

    private static func write(_ body: () -> String) {
        print("\(tag) \(body())")
    }

    package static func verbose(_ message: @autoclosure () -> String) {
        guard currentLevel <= .verbose else { return }
        write { message() }
    }

    package static func debug(_ message: @autoclosure () -> String) {
        guard currentLevel <= .debug else { return }
        write { message() }
    }

    package static func info(_ message: @autoclosure () -> String) {
        guard currentLevel <= .info else { return }
        write { message() }
    }

    package static func warn(_ message: @autoclosure () -> String) {
        guard currentLevel <= .warn else { return }
        write { message() }
    }

    package static func error(_ message: @autoclosure () -> String) {
        guard currentLevel <= .error else { return }
        write { message() }
    }

    /// Backward-compatible convenience that logs at INFO level.
    package static func log(_ message: String) {
        info(message)
    }
}
